Tests/PackageGraphPerformanceTests/DependencyResolverPerfTests.swift

```diff
@@ -0,0 +1,154 @@
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [MockPackageConstraint(container: "A", versionRequirement: v1Range)])
+                XCTAssertEqual(result, ["A": v1, "B": v1, "C": v1, "D": v1])
+            }
+        }
+    }
+
+    func testResolutionWith100Depth0Breadth_X1() {
+        let N = 1
+        let depth = 100
+        let breadth = 0
+
+        let graph = createDummyGraph(depth: depth, breadth: breadth)
+        let resolver = MockDependencyResolver(MockPackagesProvider(containers: graph.containers), MockResolverDelegate())
+
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [graph.rootConstraint])
+                XCTAssertEqual(result[0].version, v1)
+            }
+        }
+    }
+
+    func testResolutionWith100Depth1Breadth_X1() {
+        let N = 1
+        let depth = 100
+        let breadth = 1
+
+        let graph = createDummyGraph(depth: depth, breadth: breadth)
+        let resolver = MockDependencyResolver(MockPackagesProvider(containers: graph.containers), MockResolverDelegate())
+
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [graph.rootConstraint])
+                XCTAssertEqual(result[0].version, v1)
+            }
+        }
+    }
+
+    func testResolutionWith1Depth100Breadth_X1() {
+        let N = 1
+        let depth = 1
+        let breadth = 100
+
+        let graph = createDummyGraph(depth: depth, breadth: breadth)
+        let resolver = MockDependencyResolver(MockPackagesProvider(containers: graph.containers), MockResolverDelegate())
+
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [graph.rootConstraint])
+                XCTAssertEqual(result[0].version, v1)
+            }
+        }
+    }
+
+    func testResolutionWith10Depth20Breadth_X1() {
+        let N = 1
+        let depth = 10
+        let breadth = 20
+
+        let graph = createDummyGraph(depth: depth, breadth: breadth) 
+        let resolver = MockDependencyResolver(MockPackagesProvider(containers: graph.containers), MockResolverDelegate())
+
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [graph.rootConstraint])
+                XCTAssertEqual(result[0].version, v1)
+            }
+        }
+    }
+}
+
+/// Create dummpy graph with depth X breadth nodes.
+///
+/// This method creates a graph with `depth` number of main nodes. Each main node is dependent on the next main node and
+/// `breadth` number of child nodes. A root node created is dependent on the first main node.
+///
+/// - Parameters:
+///   - depth: The number of main nodes.
+///   - breadth: The number of child nodes dependent on each main node.
+/// - Returns: A tuple with root constaint and the containers.
```

Comment:
I think it would be good to describe the graph structure a little more here... I was actually expecting a tree where breadth was the branching factor at each node (so if breadth = 2 it would make a complete binary tree).

This graph does seem useful too, since you can easily calculate the number of nodes in it, but it would just be good to describe it well.